apps/cms.c

```diff
@@ -830,19 +830,18 @@ int cms_main(int argc, char **argv)
         }
 
         if (*argv != NULL) {
```

Comment:
When you're fixing it please also augment the commit message to be more descriptive. For example: `apps/cms.c: Fix unreachable code in cms_main()`

Reply:
I fix commit and moved the output of the Warning message to the code block where all the other warnings are located.
I also fix the commit message as you suggested.

Reply:
But you still kept the redundant `if(*argv != NULL)` here.

Reply:
> This if is redundant as well. It's the same condition as in the for loop.

Yes, it's the same condition as in the for loop, but that's because there's an else clause to go with it - it wasn't redundant, and looks like trying to handle it has led to a bug (unless I'm missing something, which is of course possible)

Reply:
Thinking further about this, I'd like the `argv` check and `BIO_printf()` returned here - i.e. that part was better the way it way, because now there is tight coupling between the code here and the code at line 805 (so for example, if the `*argv != NULL` check was changed to further restrict what would be valid certificates, line 805would also need to be changed.

Reply:
I don't think using `*argv != NULL` is redundant, I want to fix unreachable code with warning under else

If I remove the use of `if (*argv != NULL)` twice, then I have to add another check `if (operation == SMIME_ENCRYPT)`

```
		if (*argv != NULL) {
			if (operation == SMIME_ENCRYPT) {
			    for (; *argv != NULL; argv++) {
                cert = load_cert(*argv, FORMAT_UNDEF,
                                 "recipient certificate file");
                if (cert == NULL)
                    goto end;
                sk_X509_push(encerts, cert);
                cert = NULL;
			} else {
				BIO_printf(bio_err, "Warning: recipient certificate file parameters ignored for operation other than -encrypt\n");
			}
		}
		
		if (operation == SMIME_ENCRYPT) {
		        if (!cipher) {
#ifndef OPENSSL_NO_DES
            cipher = (EVP_CIPHER *)EVP_des_ede3_cbc();
#else
            BIO_printf(bio_err, "No cipher selected\n");
            goto end;
#endif
        }

        if (secret_key && !secret_keyid) {
            BIO_printf(bio_err, "No secret key id\n");
            goto end;
        }
		}

```

Reply:
> @nv-dmd You do not need to add anything.
> 
> Just remove the `if (*argv != NULL) {}` around the `for (; *argv != NULL; argv++) {}`. It is obviously redundant!

Thank you. You're right. I didn't think about it